site/docs/master/csi.md

```diff
@@ -2,13 +2,26 @@
 
 _This feature is under development. Documentation may not be up-to-date and features may not work as expected._
 
-Velero supports taking Container Storage Interface (CSI) snapshots as a beta feature on clusters that meet the following prerequisites.
+# Prerequisites
+
+The following are the prerequisites for using Velero to take Container Storage Interface (CSI) snapshots:
 
  1. The cluster is Kubernetes version 1.17 or greater.
  1. The cluster is running a CSI driver capable of support volume snapshots at the [v1beta1 API level](https://kubernetes.io/blog/2019/12/09/kubernetes-1-17-feature-cis-volume-snapshot-beta/).
  1. The Velero server is running with the `--features EnableCSI` feature flag to enable CSI logic in Velero's core.
  1. The Velero [CSI plugin](https://github.com/vmware-tanzu/velero-plugin-for-csi/) is installed to integrate with the CSI volume snapshot APIs.
- 1. When restoring CSI volumesnapshots across clusters, the name of the CSI driver in the destination cluster should be the same as that on the source cluster to ensure cross cluster portability of CSI volumesnapshots
+ 1. When restoring CSI volumesnapshots across clusters, the name of the CSI driver in the destination cluster is the same as that on the source cluster to ensure cross cluster portability of CSI volumesnapshots
+
+# Implementation Choices
+
+This section documents some of the choices made during implementation of the Velero [CSI plugin](https://github.com/vmware-tanzu/velero-plugin-for-csi/):
+
+1. Volumesnapshots created by the plugin will be retained only for the lifetime of the backup even if the `DeletionPolicy` on the volumesnapshotclass is set to `Retain`. This will happen during deletion of the backup.
```

Comment:
It would be helpful to explain that in order to do this, we flip the deletion policy to "Delete" just before deletion, which triggers the VSC/storage snapshot deletion.

Reply:
✅